event.c

```diff
@@ -676,17 +676,17 @@ static void event_image_preview(struct context *cnt,
 
         if ((cnt->ffmpeg_output || (cnt->conf.movie_extpipe_use && cnt->extpipe)) && !use_imagepath) {
             if (cnt->conf.movie_extpipe_use && cnt->extpipe) {
-                basename_len = strlen(cnt->extpipefilename) + 1;
-                strncpy(previewname, cnt->extpipefilename, basename_len);
+                basename_len = MIN(strlen(cnt->extpipefilename) + 1, sizeof(previewname) - 1);
+                strncpy(previewname, cnt->extpipefilename, sizeof(previewname));
                 previewname[basename_len - 1] = '.';
             } else {
                 /* Replace avi/mpg with jpg/ppm and keep the rest of the filename. */
-                basename_len = strlen(cnt->newfilename) - 3;
-                strncpy(previewname, cnt->newfilename, basename_len);
+                basename_len = MIN((cnt->newfilename - strrchr(cnt->newfilename, '.')) + 1, sizeof(previewname) - 1);
+                strncpy(previewname, cnt->newfilename, sizeof(previewname));
             }
 
             previewname[basename_len] = '\0';
-            strcat(previewname, imageext(cnt));
+            strncat(previewname, imageext(cnt), sizeof(previewname) - basename_len - 1);
```

Comment:
These two lines (`previewname` and `strncat`) are now unnecessary and be completely removed.

Furthermore, since we're no longer using `basename_len`, you can delete the declaration on line 661: https://github.com/Motion-Project/motion/blob/52ac08c6868a7a597d401ad7ad7f8de5dfebc2a0/event.c#L661